core/src/test/java/org/apache/iceberg/TestRowDelta.java

```diff
@@ -1414,18 +1415,69 @@ public void testRowDeltaCaseSensitivity() {
   }
 
   @Test
-  public void testRowDeltaToBranchUnsupported() {
-    AssertHelpers.assertThrows(
-        "Should reject committing row delta to branch",
-        UnsupportedOperationException.class,
-        "Cannot commit to branch someBranch: org.apache.iceberg.BaseRowDelta does not support branch commits",
-        () ->
+  public void testBranchConflictingDeletes() {
```

Comment:
I think that this class either needs to copy all of the tests and validate against a branch (like `testValidateDataFilesExistDefaults`), or be parameterized. I prefer parameterized.

If the tests are parameterized for `main` and `some_branch`, you could refactor all of the `commit` calls to something like this:

```
  public void commitOperation(SnapshotProducer producer) {
    if (branch.equals("main")) {
      producer.commit();
    } else {
      producer.toBranch(branch).commit();
    }
  }
```